### PR TITLE
Tpetra: Remove "DynamicProfile" mentions from Doxygen documentation

### PR DESCRIPTION
**Pull request by mhoemmen**
@trilinos/tpetra 

- Remove references to DynamicProfile from CrsGraph and CrsMatrix documentation
- Remove redundant initialization
- Remove a function that PR #6845 made redundant
- Make more CrsMatrix debug checks a run-time choice
- De-template some CrsMatrix methods that didn't need to be templated

## Motivation

Y'all thought it was a good idea to remove DynamicProfile from CrsGraph and CrsMatrix; this commit finishes the process by updating the documentation.

## Related Issues

* Related to #4701, #4755, #6655 

## Testing

Mac Clang, OpenMPI.

### CODE REVIEW

File: packages/tpetra/core/src/Tpetra_CrsGraph_decl.hpp [hunk 1]
```diff
@@ -123,17 +123,11 @@ namespace Tpetra {
     /// \brief Status of the graph's or matrix's storage, when not in
     ///   a fill-complete state.
     ///
-    /// When a CrsGraph or CrsMatrix is <i>not</i> fill complete, its
-    /// data live in one of three storage formats:
+    /// When a CrsGraph or CrsMatrix is <i>not</i> fill complete and
+    /// is allocated, then its data live in one of three storage
```
Comment (brian-kelley) [inline packages/tpetra/core/src/Tpetra_CrsGraph_decl.hpp]: There are only 2 formats now

Comment (mhoemmen) [reply packages/tpetra/core/src/Tpetra_CrsGraph_decl.hpp]: @brian-kelley Nice catch! : - D  I'll fix this first.

**Comment by mhoemmen** [comment]
One warning, in the warnings-as-errors build, that wasn't in my build.  It looks easy to fix.
```
.../packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp: In instantiation of ‘Tpetra::CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>::local_ordinal_type Tpetra::CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>::replaceLocalValues(Tpetra::CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>::local_ordinal_type, const Kokkos::View<const LocalOrdinal*, Kokkos::AnonymousSpace>&, const Kokkos::View<const typename Kokkos::Details::ArithTraits<RealFloatType>::val_type*, Kokkos::AnonymousSpace>&) const [with Scalar = long long int; LocalOrdinal = int; GlobalOrdinal = long long int; Node = Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial>; Tpetra::CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>::local_ordinal_type = int; typename Kokkos::Details::ArithTraits<RealFloatType>::val_type = long long int]’:
packages/tpetra/coreTpetra_CrsMatrix_LONG_LONG_INT_LONG_LONG_SERIAL.cpp:76:3:   required from here
.../packages/tpetra/coreTpetra_CrsMatrix_def.hpp:2439:21: error: comparison of integer expressions of different signedness: ‘const LO’ {aka ‘const int’} and ‘std::enable_if<true, long unsigned int>::type’ {aka ‘long unsigned int’} [-Werror=sign-compare]
     if (numInputEnt != inputVals.extent(0)) {
cc1plus: all warnings being treated as errors
```

**Comment by mhoemmen** [comment]
Thanks Brian! : - D

**Pull request merged**